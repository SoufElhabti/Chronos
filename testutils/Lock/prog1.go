package Lock

import (
	"sync"
)

var a int

func fn1() {
	mutex := sync.Mutex{}
	mutex.Lock()
	a = 5
}
