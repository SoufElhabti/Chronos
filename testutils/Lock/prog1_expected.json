{
	"Lockset": {
		"ExistingLocks": {
			"t01598535": 1598568
		},
		"ExistingUnlocks": {}
	},
	"GuardedAccess": [
		{
			"ID": 1,
			"Pos": 1598572,
			"Value": 0,
			"OpKind": 0,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 1
				},
				"LocksetJson": {
					"ExistingLocks": {
						"t01598535": 1598568
					},
					"ExistingUnlocks": {}
				}
			}
		},
		{
			"ID": 2,
			"Pos": 1598572,
			"Value": 1598513,
			"OpKind": 1,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 2
				},
				"LocksetJson": {
					"ExistingLocks": {
						"t01598535": 1598568
					},
					"ExistingUnlocks": {}
				}
			}
		}
	]
}